key/src/main/java/software/amazon/kms/key/BaseHandlerStd.java

```diff
@@ -156,4 +158,23 @@ protected void resourceStateCheck(final KeyMetadata keyMetadata) {
           return ProgressEvent.defaultFailureHandler(e, HandlerErrorCode.NotFound);
       return ProgressEvent.defaultFailureHandler(e, HandlerErrorCode.GeneralServiceException);
     }
+
+    /**
+     * A helper method for validating that the requested resource model transition is possible
+     */
+    protected static ProgressEvent<ResourceModel, CallbackContext> validateResourceModel(
+            final ProgressEvent<ResourceModel, CallbackContext> progress, final ResourceModel previousModel,
+            final ResourceModel model) {
+        // If key is not being updated and is disabled, then we cannot perform any updates on the key
```

Comment:
probably not `any updates` but rotation updates?

Reply:
Re-worded this comment